addons/website/static/src/js/content/menu.js

```diff
@@ -660,6 +657,130 @@ publicWidget.registry.hoverableDropdown = animations.Animation.extend({
     },
 });
 
+publicWidget.registry.megaMenuDropdown = publicWidget.Widget.extend({
+    selector: "header#top",
+    disabledInEditableMode: false,
+    events: {
+        "mousedown .o_mega_menu_toggle": "_onMegaMenuClick",
+        "mouseenter .o_mega_menu_toggle": "_onMegaMenuHover",
+        "mousedown .o_extra_menu_items": "_onExtraMenuClick",
```

Comment:
Apparently for some reason, the ArrowUp key also works to open dropdowns (you might want to dig in to find the source of that behavior, maybe there are more keys that work that should be cared for).

To be clear about the problem here:
- Open the mega menu on desktop
- Toggle the mobile preview
- Open the mobile menu
- Navigate with `Tab` up to the mega menu, and press one of the supported keys
=> traceback + the menu is corrupted.

So if you forget any key, there is a chance to produce that error.

Reply:
Argh, I didn't think about the keys, I didn't even know we should support them :grimacing: 
I will adapt, thanks for telling me :slightly_smiling_face:

Reply:
I added `keyup` listeners to handle that :wink: Note that using tab causes a traceback in edit mode in general (I will create a task for that)
I tested on my phone and there doesn't seem to be any issues with touch events :thinking:

Reply:
I see you didn't set the `focus` state.
I know it's not implemented right now, but I'd like to insist on that. :) It's wrong that it's not implemented, and we should take this opportunity to fix that.
From an accessibility perspective, an action on `hover` translates into `focus`. It would probably be ok-ish if there was no visual difference between a menu that opens on hover and one that opens on click, but there is: the `:focus()` state is not represented the same way. On click, the `:focus()` is a well-marked thick outline. On hover, the `:focus()` is a text with a dimmed grey and no outline: it isn't clear/not very visible what you're on if you tab on it.
That makes perfect sense for a click/hover distinction. But then if we want to make it accessible, the behavior should be on par with mouse navigation: open on focus.

Reply:
Found the cause of this issue: in fact the focus works fine, the outline is just not displayed because of the rule added [here](https://github.com/odoo/odoo/commit/5846a05fe7bf587d3bfb3ace3e863cb4abd52132#diff-f338780f594d6b1f46eda5d3fb9fed86f3c3c8f65522fb93e639534e9d92cdb2R815-R818) :grimacing: It was removed because it was also displayed when hovering with the mouse 

Since it works fine before this rule (in version 15), I will make a proper fix in 16.0 to allow displaying the outline when it is focused with the tab key :slightly_smiling_face: (so not in this PR)